### PR TITLE
Issue/2290 refund shipping labels

### LINKED ISSUE #2290: Shipping Labels M1 release: add option to refund a shipping label
If shipping labels are available for an order, we should provide an option to refund an already purchased shipping label. **Note that this [FluxC issue](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/1557) should be addressed first.**

screenshot-2020-04-21-at-6 32 44-pm

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #2290 by providing the option to refund a shipping label.

#### Screenshots
. 

#### Note
- **Since shipping labels are tested against a staging WCS, please note that we need to refund a shipping label within 2 minutes of creating it. Otherwise, the refund API results in an error, whilst the refund itself is successful**
- **To create shipping labels, please follow the instructions provided in the Shipping Labels Master Thread - p91TBi-2om**
- The changes in this PR is only enabled for debug users and is behind a feature flag.

#### To test
- Click on the ellipsis icon of a shipping label item of an order.
- Click on the **Request a refund** button.
- Notice that the shipping label amount and date are displayed correctly.
- Click on the Refund button.
- Notice that a Snackbar message is displayed when the refund is initiated.
- If the refund us successful, notice a SnackBar message is displayed and you are redirected to the Order detail screen. Notice the shipping label is updated to reflect the correct refunded status.
- If the refund results in an error, notice a snackbar message is displayed. **In order to initiate an error scenario, create a shipping label and wait atleast 5 minutes before refunding the label from the app.** 

While the above testing steps are applicable for all orders with shipping labels, it would be great to test those steps for the below scenarios:
- Try refunding a shipping label for a single package shipping label order that is fulfilled already i.e. order status is **Completed**.  
- Try refunding a shipping label for a single package shipping label order that is not yet fulfilled already i.e. order status is **Processing**.  
- Try refunding a shipping label for a multi package shipping label order that is fulfilled already i.e. order status is **Completed**.  
- Try refunding a shipping label for a multi package shipping label order that is not yet fulfilled already i.e. order status is **Processing**. 
- Try refunding a shipping label with the Shipment Tracking plugin activated/deactivated. This should not affect the refund process but is added as part of the testing process to cover all scenarios.
- Try refunding a shipping label with the order status **Refunded**. Again, the order status should not affect the refund process but is added as part of the testing process to cover all scenarios. 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/ShippingLabelRefundViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            triggerEvent(ShowSnackbar(string.shipping_label_refund_progress_message))
+
+            refundJob = launch {
+                val result = repository.refundShippingLabel(arguments.orderId, arguments.shippingLabelId)
+                if (result.isError) {
+                    // TODO: add tracking event to track the failure when refunding shipping label
+                    triggerEvent(ShowSnackbar(string.order_refunds_amount_refund_error))
+                } else {
+                    // TODO: add tracking event to track the success when refunding shipping label
+                    triggerEvent(ShowSnackbar(string.shipping_label_refund_success))
```
Comment (ThomazFB) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/ShippingLabelRefundViewModel.kt:54]: Just a quick question, are those TODOs for track events something to be done on another task or they are missing here?

**Comment by ThomazFB** [comment]
Hey @anitaa1990, great work here! I've left just one comment about tracks event, but if the comment doesn't make sense, just ignore it, also, there's a conflicting with a `build.gradle` file. Nevertheless, LGTM :shipit:

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/ShippingLabelRefundViewModel.kt (hunk 1 above)
Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/ShippingLabelRefundViewModel.kt:54]: Yup, all tracking events for the shipping labels feature would be taken care of in another PR 😄

**Comment by anitaa1990** [comment]
Thanks for the review @ThomazFB! I have fixed the merge conflict. PR ready for another round 👍

**Pull request merged**